bin/srcflux

```diff
@@ -1091,13 +1127,13 @@ def add_aprates_to_output( myparams, at_energy, outfiles):
 
     verb1("Adding net rates to output")
 
-    def pull_par_into_crate( parname, cratename, desc, units="counts/s" ):
+    def pull_par_into_crate( parname, cratename, desc, dtype=float, units="counts/s" ):
         from paramio import pget
         from pycrates import CrateData
         cd = CrateData()
         cd.name = cratename
         vals = [pget(pp,parname) for pp in outfiles ]
-        vals = [ np.nan if x == 'INDEF' else float(x) for x in vals ]
+        vals = [ np.nan if x == 'INDEF' else dtype(x) for x in vals ]
```

Comment:
So, NaN isn't technically valid for an array of ints. Is crates clever enough to handle this case (and is it clever enough to identify that a list of ints means we can use an integer output type, I mean I assume it is or else this change would not have been made)?

Reply:
Good catch.

Yes, crates "casts" the list to a numpy array and uses it's datatype, and numpy is smart enough to check all the values and if they're all int's will create an int column.  If there'd happen to be a NaN, then it'd get cast as a float ... so not what we want.

We'd have to set a NULL value and deal w/ it that way.